pkg/state/eth_info.go

```diff
@@ -0,0 +1,107 @@
+package state
+
+import (
+	"github.com/pkg/errors"
+	"github.com/wavesplatform/gowaves/pkg/errs"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+	"github.com/wavesplatform/gowaves/pkg/proto/ethabi"
+	"github.com/wavesplatform/gowaves/pkg/settings"
+)
+
+const (
+	EthereumTransferWavesKind = iota + 1
+	EthereumTransferAssetsKind
+	EthereumInvokeKind
+)
+
+type ethInfo struct {
+	stor     *blockchainEntitiesStorage
+	settings *settings.BlockchainSettings
+}
+
+func newEthInfo(stor *blockchainEntitiesStorage, settings *settings.BlockchainSettings) *ethInfo {
+	return &ethInfo{stor: stor, settings: settings}
+}
+
+func GuessEthereumTransactionKind(
+	data []byte,
+	to *proto.EthereumAddress,
+	newestAssetInfo func(id proto.AssetID, filter bool) (*assetInfo, error)) (int64, error) {
+	if data == nil {
+		return EthereumTransferWavesKind, nil
+	}
+
+	selectorBytes := data
+	if len(data) < ethabi.SelectorSize {
+		return 0, errors.Errorf("length of data from ethereum transaction is less than %d", ethabi.SelectorSize)
+	}
+	selector, err := ethabi.NewSelectorFromBytes(selectorBytes[:ethabi.SelectorSize])
+	if err != nil {
+		return 0, errors.Errorf("failed to guess ethereum transaction kind, %v", err)
```

Comment:
Use `errors.Wrap`

Reply:
what's the difference?

Reply:
Wrap saves stack info, Errorf creates new error with empty stack info